scipy/stats/setup.py

```diff
@@ -1,5 +1,10 @@
 from os.path import join
 
+def pre_build_hook(build_ext, ext):
```

Comment:
`pre_build_hook` is used to add C++11/14 flag for the compiler (if you need it)

Reply:
I don't know what a hook is, but I assume from the context it's something that needs to be added to get support for different C++ standards?

What C++ standards do I get if I don't add the hook?

I'm using g++, I think. Whatever I got when I setup my Windows dev environment by blindly following [these instructions](http://scipy.github.io/devdocs/building/windows.html).

And I don't know anything about modern C++ - I'm guessing I need these if the code is trying to take advantage of new C++ features?

Reply:
I understand a "hook" to be similar to a callback.  Different hooks run after different things -- in this case, the `pre_build_hook` runs before the build_ext step when building a C extension.  Inside the hook function have a chance to change compiler flags and other odds and ends.  Since compiler flags are in general not portable between compilers, scipy has some hard coded flags based on the compilers it supports.

The C++ standard you get by default varies by compiler.  You will almost certainly have the C++03 standard which is a revision of the C++98 standard.  Almost all C++ legacy code encountered in the wild will be C++98 conformant.  Newer clang/g++ compilers will support C++11 out of the box, but to make sure everyone has those features enabled you want to pass the flags (we ran into this trouble when compiling HiGHS).  I believe C++14 is not yet available on all platforms where scipy is expected to build, so C++14 specific features should be avoided for now (e.g., `std::make_unique`).

This example doesn't need any modern C++ features, so it would be fine to leave it out.  I just wanted to mention how you enable modern features if you needed them when you wrap that stats functions.